src/jdk.jlink/share/classes/jdk/tools/jlink/internal/plugins/SystemModulesPlugin.java

```diff
@@ -560,42 +562,38 @@ static class SystemModulesClassGenerator {
         // A builder to create one single Set instance for a given set of
         // names or modifiers to reduce the footprint
         // e.g. target modules of qualified exports
-        private final DedupSetBuilder dedupSetBuilder
-            = new DedupSetBuilder(this::getNextLocalVar);
+        private final DedupSetBuilder dedupSetBuilder;
 
         public SystemModulesClassGenerator(String className,
                                            List<ModuleInfo> moduleInfos,
                                            int moduleDescriptorsPerMethod) {
             this.classDesc = ClassDesc.ofInternalName(className);
             this.moduleInfos = moduleInfos;
             this.moduleDescriptorsPerMethod = moduleDescriptorsPerMethod;
+            this.dedupSetBuilder = new DedupSetBuilder(this.classDesc);
             moduleInfos.forEach(mi -> dedups(mi.descriptor()));
         }
 
-        private int getNextLocalVar() {
-            return nextLocalVar++;
-        }
-
         /*
          * Adds the given ModuleDescriptor to the system module list.
          * It performs link-time validation and prepares mapping from various
          * Sets to SetBuilders to emit an optimized number of sets during build.
          */
         private void dedups(ModuleDescriptor md) {
             // exports
-            for (Exports e : md.exports()) {
+            for (Exports e : sorted(md.exports())) {
```

Comment:
Is sorting needed?

Reply:
I will need to check again, this was added when we need to ensure the sequence of cache from the dedupSet.

Reply:
Delay the sorting to the DedupSet build time.